diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1057a6f..dd0e266 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -137,6 +137,10 @@ target_include_directories(glfw PRIVATE
                            "${GLFW_BINARY_DIR}/src")
 target_link_libraries(glfw PRIVATE Threads::Threads)

+if (CMAKE_SYSTEM_NAME MATCHES "QNX")
+    target_link_libraries(glfw PUBLIC -ltimerfd)
+endif()
+
 # Workaround for CMake not knowing about .m files before version 3.16
 if (CMAKE_VERSION VERSION_LESS "3.16" AND APPLE)
     set_source_files_properties(cocoa_init.m cocoa_joystick.m cocoa_monitor.m

